playground/backend/internal/cache/local/local_cache.go

```diff
@@ -144,6 +148,22 @@ func (lc *Cache) GetDefaultPrecompiledObject(ctx context.Context, sdk pb.Sdk) (*
 	return defaultPrecompiledObject, nil
 }
 
+func (lc *Cache) SetSdkCatalog(_ context.Context, sdks []*entity.SDKEntity) error {
+	lc.Lock()
```

Comment:
Do not embed the mutex on the struct. Probably it was already there before you, but maybe it is a good point to fix it?